pandas/core/indexes/datetimes.py

```diff
@@ -2030,7 +2031,7 @@ def date_range(start=None, end=None, periods=None, freq='D', tz=None,
 
     Notes
     -----
-    2 of start, end, or periods must be specified
+    Exactly two of start, end, or periods must be specified
```

Comment:
I know what you're trying to say here, but it's a little confusing the final parameter is plural.  Say something like this (and propagate to places where relevant):

"Of the three parameters, start, end, and periods, exactly two must be specified."

For the doc-strings, I would even consider ticking them for additional clarity.